src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -476,21 +476,31 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
     if (stateObj?.state !== UNAVAILABLE) {
       return;
     }
-    const validation = await validateConfig(this.hass, {
-      trigger: this._config.trigger,
-      condition: this._config.condition,
-      action: this._config.action,
-    });
-    this._validationErrors = (
-      Object.entries(validation) as Entries<typeof validation>
-    ).map(([key, value]) =>
-      value.valid
-        ? ""
-        : html`${this.hass.localize(
-              `ui.panel.config.automation.editor.${key}s.name`
-            )}:
-            ${value.error}<br />`
-    );
+
+    try {
+      const validation = await validateConfig(this.hass, {
+        trigger: this._config.trigger,
+        condition: this._config.condition,
+        action: this._config.action,
+      });
+
+      this._validationErrors = (
+        Object.entries(validation) as Entries<typeof validation>
+      ).map(([key, value]) =>
+        value.valid
+          ? ""
+          : html`${this.hass.localize(
+                `ui.panel.config.automation.editor.${key}s.name`
+              )}:
+              ${value.error}<br />`
+      );
+    } catch (errors: any) {
+      this._validationErrors = errors.message || errors.code;
+      showToast(this, {
```

Comment:
I don't think we should show a toast here? We already display this error message?

Reply:
I wondered this too but that’s how the existing functionality works and I modelled it off that. If we change one then we should change both so they are consistent.

Reply:
I dont see a toast being called in the code outside the catch?

Reply:
It’s more or less identical to this:

https://github.com/home-assistant/frontend/blob/d4386b5e8cc5d354c612fe31a65338629c9b5bb0/src/panels/config/automation/ha-automation-editor.ts#L736

Reply:
That makes sense. I thought not having toast may be an issue on mobile but as it’s only on load, shouldn’t be a problem.

Reply:
Main reason for that (I think) is that on save you can be scrolled down, and can miss the error on the top, on load you will always be at the top of the screen and see the error message.

Reply:
Ah, perfect. I will make that change then.